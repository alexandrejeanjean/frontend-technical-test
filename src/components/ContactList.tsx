import type { FC } from 'react'
import Contact from './contact'

const List = [{ name: 'Alexandre Jeanjean' }, { name: 'OcÃ©ane Olmi' }]

const ContactList: FC = () => {
  return (
    <ul className='flex flex-col'>
      {List.map((contact, id) => (
        <Contact key={'contact_' + id} name={contact.name} />
      ))}
    </ul>
  )
}

export default ContactList
