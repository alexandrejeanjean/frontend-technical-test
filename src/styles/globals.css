@tailwind base;
@tailwind components;
@tailwind utilities;

.arrow {
  @apply block p-1 border-t-0 border-b-2 border-l-0 border-r-2 border-gray-500 border-solid;
}

.right {
  transform: rotate(-45deg);
  -webkit-transform: rotate(-45deg);
  @apply border-indigo-600;
}

.main-container-height {
  height: calc(100vh - 132px);
}

.list-height {
  height: calc(100% - 50px);
}