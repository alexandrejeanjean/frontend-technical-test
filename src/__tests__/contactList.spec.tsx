import { expect, jest } from '@jest/globals';
import { shallow } from 'enzyme';
import ContactList from '../components/ContactList';


const handleConversationSelection = jest.fn();

/** @test {ContactList Component} */
describe('ContactList Component', () => {

    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<ContactList handleConversationSelection={handleConversationSelection} />);
    });

    it('should render without crashing', () => {
        expect(wrapper.find('ul')).toHaveLength(1);
    });

    it('match snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    })

});

