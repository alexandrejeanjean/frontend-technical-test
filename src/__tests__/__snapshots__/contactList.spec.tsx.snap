// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactList Component match snapshot 1`] = `ShallowWrapper {}`;
